### PR TITLE
Metrics: Added allowlist filtering

### PR DESCRIPTION
**Pull request by eloycoto**
With this commit, users can define what metrics want to to retrieve
on workload level.

Related-to: https://github.com/project-flotta/flotta-operator/commit/5b67dd38215b6a110a6293c9ac35875e9417daa0
Fix: ECOPROJECT-495

Signed-off-by: Eloy Coto <eloy.coto@acalustra.com>

### CODE REVIEW

File: internal/metrics/workload.go [hunk 1]
```diff
@@ -68,6 +68,11 @@ func (wrkM *WorkloadMetrics) WorkloadStarted(workloadName string, report []*podm
 			continue
 		}
 
+		var filter SampleFilter = &PermissiveAllowList{}
+		if cfg.Metrics.AllowList != nil && len(cfg.Metrics.AllowList.Names) > 0 {
```
Comment (eloycoto) [inline internal/metrics/workload.go:72]: @jakub-dzon  not sure here, if there are no Names defined should we do Restrictive or Permissive.

Comment (jakub-dzon) [reply internal/metrics/workload.go:72]: IMO when it's nil, it should be Permissive, if empty - Restrictive.

File: internal/metrics/workload.go [hunk 2]
```diff
@@ -78,7 +83,11 @@ func (wrkM *WorkloadMetrics) WorkloadStarted(workloadName string, report []*podm
 			interval = cfg.Metrics.Interval
 		}
 		// log for this is part of the AddTarget function
-		wrkM.daemon.AddTarget(workload.Name, urls, time.Duration(interval)*time.Second)
+		wrkM.daemon.AddFilteredTarget(
```
Comment (jakub-dzon) [inline internal/metrics/workload.go:86]: Can we remove `AddTarget` at this point?

File: internal/metrics/workload.go (hunk 1 above)
Comment (eloycoto) [reply internal/metrics/workload.go]: :+1:  I was not sure about.

File: internal/metrics/workload.go (hunk 2 above)
Comment (eloycoto) [reply internal/metrics/workload.go]: Removed the AddFilteredTarget and rename to AddTarget.

File: internal/metrics/workload.go [hunk 3]
```diff
@@ -68,6 +68,11 @@ func (wrkM *WorkloadMetrics) WorkloadStarted(workloadName string, report []*podm
 			continue
 		}
 
+		var filter SampleFilter = &PermissiveAllowList{}
```
Comment (jakub-dzon) [inline internal/metrics/workload.go:71]: `filter SampleFilter := &PermissiveAllowList{}`

**Review by jakub-dzon — APPROVED** [review]

**Review by masayag — APPROVED** [review]

**Pull request merged**